.github/workflows/build-and-test.yml

```
@@ -0,0 +1,29 @@
+name: Devops Course Todo List Pipeline
+on: 
+  push:
+    paths-exclude:
+      - 'util/**'
+      - 'README.md'
+      - 'Makefile'
+      - 'hooks/**'
+      - 'reading/**'
+  pull_request:
+    branches: 'master'
+
+jobs:
+  build:
+    name: Unit Tests 
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout 
+      uses: actions/checkout@v2
+      timeout-minutes: 1
+    - name: Touch .env  
+      run: touch .env
+      timeout-minutes: 1
+    - name: Prepare
+      run: docker compose build test
+      timeout-minutes: 1
+    - name: Test
+      run: docker compose run test
+      timeout-minutes: 1
```

No strong preference for me, but worth noting that we can combine these:

```suggestion
    - name: Test
      run: docker compose run --build test
      timeout-minutes: 1
```

Note that it's certainly plausible this hits the 1 minute timeout for me if our image grows, but maybe that's still an important feature; we can easily disable it later if needed, but it will give us a natural quick to try wrestle the times back down if so!